### PR TITLE
feat: use web components 21.0.0-alpha2, fix tests

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

1. Updated the web components to new major
2. Added the `@vaadin/vaadin-template-renderer` to bring back removed Polymer template APIs.

### CODE REVIEW

**Review by ZheSun88 — APPROVED** [review]

**Comment by web-padawan** [comment]
Remaining failures:

```
javascript error: arguments[0]._scrollToIndex is not a function
```

```
java.lang.AssertionError: A grid cell was expected to have text content but had none.
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at com.vaadin.flow.component.grid.it.GridTestScrollingOver100kLinesIT.lambda$toStringIsUsedForObjectSerialization$1(GridTestScrollingOver100kLinesIT.java:53)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at com.vaadin.flow.component.grid.it.GridTestScrollingOver100kLinesIT.toStringIsUsedForObjectSerialization(GridTestScrollingOver100kLinesIT.java:49)
```

```
java.lang.NullPointerException
	at com.vaadin.flow.component.grid.testbench.GridElement.getFirstVisibleRowIndex(GridElement.java:72)
	at com.vaadin.flow.component.grid.testbench.GridElement.isRowInView(GridElement.java:174)
	at com.vaadin.flow.component.grid.testbench.GridElement.getCell(GridElement.java:115)
	at com.vaadin.flow.component.grid.testbench.GridElement.getCell(GridElement.java:100)
	at com.vaadin.flow.component.grid.it.RecalculateColumnWidthsIT.columnsRecalculateAfterDataChange(RecalculateColumnWidthsIT.java:40)
```

And the one about `lit-element` import:

```
java.lang.AssertionError: Expected only [Deprecation] warnings in the logs
Expected: a string containing "HTML Imports"
     but: was "http://172.16.1.126:8080/VAADIN/build/vaadin-1-0efec97cdf48f1cc4870.cache.js 507:8 "The main 'lit-element' module entrypoint is deprecated. Please update your imports to use the 'lit' package: 'lit' and 'lit/decorators.ts' or import from 'lit-element/lit-element.ts'.""
```

**Comment by web-padawan** [comment]
Few other failures:

```
java.lang.AssertionError: Item count doesn't match expected:<1501> but was:<1800>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at com.vaadin.flow.component.grid.dataview.AbstractItemCountGridIT.verifyRows(AbstractItemCountGridIT.java:99)
	at com.vaadin.flow.component.grid.dataview.ItemCountEstimateGridIT.itemCountEstimateGrid_estimateLessThanCurrentRange_estimateNotChanged(ItemCountEstimateGridIT.java:102)
```

```
java.lang.AssertionError: Grid should be able to scroll after changing to defined size expected:<517> but was:<516>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at com.vaadin.flow.component.grid.dataview.ItemCountUnknownGridIT.undefinedSizeGrid_switchesToDefinedSize_sizeChanges(ItemCountUnknownGridIT.java:84)
```

```
org.junit.ComparisonFailure: Cell content should have not updated expected:<[EDITED]> but was:<[0]>
	at org.junit.Assert.assertEquals(Assert.java:117)
	at com.vaadin.flow.component.grid.it.GridItemRefreshPageIT.dataProviderRefreshItem_gridDoesntTouchOtherItemsOnDOM(GridItemRefreshPageIT.java:92)
```

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/dataview/ItemCountEstimateGridIT.java [hunk 1]
```diff
@@ -99,10 +99,10 @@ public void itemCountEstimateGrid_estimateLessThanCurrentRange_estimateNotChange
         grid.scrollToRow(1500);
         setEstimate(1501);
 
-        verifyRows(1501);
+        verifyRows(1800);
 
-        setEstimate(1600);
-        verifyRows(1600);
+        setEstimate(1700);
+        verifyRows(1700);
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/dataview/ItemCountEstimateGridIT.java:105]: The previous tests were written with the assumption that the scrollToIndex wouldn't continue to the desired index once the last scrollable index before increasing the range has been reached. This works as expected with the new version so the grid actually ends up at index 1500 after `scrollToRow(1500)`, even if the effective size is increased dynamically.

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/dataview/ItemCountUnknownGridIT.java [hunk 2]
```diff
@@ -80,7 +80,8 @@ public void undefinedSizeGrid_switchesToDefinedSize_sizeChanges() {
 
         grid.scrollToRow(500);
 
-        int expectedLastItem = 517;
+        int VIEWPORT_ITEM_CAPACITY = 16;
+        int expectedLastItem = 500 + VIEWPORT_ITEM_CAPACITY;
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/dataview/ItemCountUnknownGridIT.java:84]: Same as https://github.com/vaadin/flow-components/pull/978/files#r632532693

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/dataview/ItemCountUnknownGridIT.java [hunk 3]
```diff
@@ -96,7 +97,7 @@ public void undefinedSizeGrid_switchesToDefinedSize_sizeChanges() {
         // size has been increased again by default size
         doScroll(1000, 1200, 6, 950, 1100);
 
-        Assert.assertEquals(999, grid.getLastVisibleRowIndex());
+        Assert.assertEquals(1000 + VIEWPORT_ITEM_CAPACITY, grid.getLastVisibleRowIndex());
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/dataview/ItemCountUnknownGridIT.java:100]: Same as https://github.com/vaadin/flow-components/pull/978/files#r632532693

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridTestScrollingOver100kLinesIT.java [hunk 4]
```diff
@@ -46,14 +46,18 @@ public void toStringIsUsedForObjectSerialization() {
...
+        List<TestBenchElement> bodySlots = grid.$("tbody slot").all();
+        Assert.assertTrue(headerSlots.size() > 0);
+        Assert.assertTrue(bodySlots.size() > 0);
+
         allCellContents.forEach(vgcc -> {
-            String slotName = vgcc.getAttribute("slot")
-                    .replace("vaadin-grid-cell-content-", "");
-            if (Integer.parseInt(slotName) <= 180) {
+            TestBenchElement slot = vgcc.getPropertyElement("assignedSlot");
+            if (headerSlots.contains(slot) || bodySlots.contains(slot)) {
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridTestScrollingOver100kLinesIT.java:56]: This test relied the cell content elements in the light DOM belonging to body and header starting from an arbitrary index 180.

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java [hunk 5]
```diff
@@ -1087,9 +1087,13 @@ private void scroll(GridElement grid, int index) {
...
-        waitUntil(driver -> getCells(grid).stream()
-                .filter(cell -> text.equals(cell.getText())).findFirst()
-                .isPresent());
+        waitUntil(driver -> {
+            List<?> cellContentTexts = (List<?>) getCommandExecutor()
+                    .executeScript(
+                            "return Array.from(arguments[0].querySelectorAll('vaadin-grid-cell-content')).map(cell => cell.textContent)",
+                            grid);
+            return cellContentTexts.contains(text);
+        });
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java:1096]: For some reason getting the cells via Selenium API resulted in different set than what querySelectorAll returns. Fixed to use pure JS approach instead.

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridVirtualScrollingIT.java [hunk 6]
```diff
@@ -33,11 +33,10 @@ public void gridScrolling() {
         waitForElementPresent(By.tagName("vaadin-grid"));
         WebElement grid = findElement(By.tagName("vaadin-grid"));
 
-        executeScript("arguments[0]._scrollToIndex(1000000);", grid);
+        executeScript("arguments[0].scrollToIndex(1000000);", grid);
 
         long firstVisible = (long) executeScript(
-                "return arguments[0]._firstVisibleIndex+arguments[0]._vidxOffset",
-                grid);
+                "return arguments[0]._firstVisibleIndex", grid);
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridVirtualScrollingIT.java:39]: `_vidxOffset` is now a concept inside the virtualizer and doesn't concern grid. Now `_firstVisibleIndex` includes the offset value (it's still not public API!)

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**